### ISSUE TITLE
Why does a url with "#" returns 404 in Heroku?

### ISSUE THREAD

**Issue by luccasmaso** [typhoeus/typhoeus#438]
I'm testing these 2 valid urls 
`http://www.businessinsider.com.au/smartphone-impact-brain-body-sleep-2015-2`
`http://www.businessinsider.com.au/smartphone-impact-brain-body-sleep-2015-2#ooid=BvMjVqcjoHdZBG6tTpXy8UkhB5_46U_c`

Running the code below for both, the first one returns 200 OK, but the second one returns 404.

```
request = Typhoeus::Request.new(url, followlocation: true)
request.on_headers do |response|
   puts response.code
end
request.run
```

Cannot have idea of this behavior. Maybe is some escaping problem witth `#`?

If I make a little replace of the # to ?, it works
`http://www.businessinsider.com.au/smartphone-impact-brain-body-sleep-2015-2?ooid=BvMjVqcjoHdZBG6tTpXy8UkhB5_46U_c`

Thanks

**Comment by luccasmaso** [comment]
Well, actually doing some more tests, this problem only occurs no production environment in my case on heroku. What heroku has to do with it?

**Comment by i0rek** [comment]
It probably has another libcurl version.

**Comment by luccasmaso** [comment]
Yes, probably it is. Found the same problem here https://github.com/typhoeus/ethon/issues/93

**Comment by felipedmesquita** [comment]
Thanks, @luccasmaso, for your report. This seems to be due to url escaping being handled differently in different libcurl versions. Since this issue is ten years old and seems to be resolved as the similar https://github.com/typhoeus/ethon/issues/93 is resolved, I'm closing it. Please don't hesitate to open a new one if you still find unexpected escaping behavior in Typhoeus. Hope you are still using it ten years latter ( :
